snuba/utils/metrics/timer.py

```diff
@@ -45,14 +46,25 @@ def finish(self) -> TimerData:
                 (name, self.__diff_ms(self.__marks[i][1], ts))
                 for i, (name, ts) in enumerate(self.__marks[1:])
             ]
+            duration_ms = self.__diff_ms(start, end)
+            duration_group = "<10s"
+            if duration_ms >= 30000:
+                duration_group = ">30s"
+            elif duration_ms >= 20000:
+                duration_group = ">20s"
+            elif duration_ms >= 10000:
+                duration_group = ">10s"
```

Comment:
Why don't you add this logic in querylog when we are actually recording the metrics?
https://github.com/getsentry/snuba/blob/master/snuba/querylog/__init__.py#L28-L35

So we do not have to add something that is specific to the performance of the query execution to an abstraction that is supposed to be generic (it was used to be used in other scenarios than query execution). Moreover adding it in the timer is redundant as we already have the duration.
You can extract the duration there and pass the duration_group as a tag.

Since you are doing the same for tracing, it should be easy to put these few lines in the same method somewhere in utils that turns a duration into a group.

Reply:
I don't think that will work if there's an error though. Do we log to querylog if there's an exception somewhere?

Reply:
We do log the query even if it failed as long as we got past the AST building stage.
https://github.com/getsentry/snuba/blob/master/snuba/web/query.py#L75-L77

It works slightly differently if the query fails before building the AST 
https://github.com/getsentry/snuba/blob/master/snuba/request/validation.py#L78-L83
In such case there is no Timer anyway, so it would not make a difference.